src/main/java/com/uid2/shared/store/parser/ServiceLinkParser.java

```diff
@@ -0,0 +1,33 @@
+package com.uid2.shared.store.parser;
+
+import com.uid2.shared.Utils;
+import com.uid2.shared.model.ServiceLink;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ServiceLinkParser implements Parser<Map<String, ServiceLink>> {
+
+    @Override
+    public ParsingResult<Map<String, ServiceLink>> deserialize(InputStream inputStream) throws IOException {
+        JsonArray spec = Utils.toJsonArray(inputStream);
+        final HashMap<String, ServiceLink> serviceLinkMap = new HashMap<>();
+        for (int i = 0; i < spec.size(); i++) {
+            JsonObject serviceLinkSpec = spec.getJsonObject(i);
+            String linkId = serviceLinkSpec.getString("link_id");
+            int serviceId = serviceLinkSpec.getInteger("service_id");
+            int siteId = serviceLinkSpec.getInteger("site_id");
+            String name = serviceLinkSpec.getString("name");
+
+            ServiceLink serviceLink = new ServiceLink(linkId, serviceId, siteId, name);
+
+            serviceLinkMap.put(linkId, serviceLink);
```

Comment:
~Is the assumption that uniqueness of the `linkId` will be checked in the upper layer i.e. on receiving the POST request and its guaranteed to be unique when it gets here ?~
Hmm..link_id here doesn't have to be unique since its not generated by us. However, the combination of <link_id, service_id> should be unique. Does that fit into the existing code pattern ?